packages/loader/driver-utils/src/parallelRequests.ts

```diff
@@ -485,8 +486,18 @@ async function getSingleOpBatch(
 				// If the error told us to wait, then we will wait for that specific amount rather than the default.
 				delay = retryAfter;
 			}
+		} finally {
+			telemetryEvent?.end({
```

Comment:
I'm not sure I follow the flow. Looking at this call, all the props are known at the end of previous iteration, so this call can be moved to like 512 just fine. And if so, no need to manually track time, as performanceEvent will do it for you.

I'm also not sure in having totalRetryAfterTime  accumulating time across multiple events (fired in this loop) - it would require devs to mostly ignore all but last event in a series, which is very hard to do.

Given that we are not that interested in how long we waited on failure path, we could actually make it a single event at the end of function, reporting sum of all wait time.

Reply:
Here I am reporting the total time we waited across several retries, however I can report only in successful case rather than in all cases. Also this event will happen only once. I am actually not sure what you are saying above.